orthoinference/src/main/java/org/reactome/orthoinference/StableIdentifierGenerator.java

```diff
@@ -0,0 +1,67 @@
+package org.reactome.orthoinference;
+
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.HashMap;
+import java.util.Map;
+import static org.gk.model.ReactomeJavaConstants.*;
+
+/*
+ *  All PhysicalEntitys, ReactionlikeEvents and Pathways are routed to this class to generate their stable identifiers
+ */
+public class StableIdentifierGenerator {
+
+    private static MySQLAdaptor dba;
+    private static GKInstance instanceEditInst;
+    private static Map<String,Integer> seenOrthoIds = new HashMap<>();
+    private static String speciesAbbreviation = null;
```

Comment:
Since the setting of `speciesAbbreviation` is required for the `generateOrthologousStableId` method, it would probably be good to extract it from the `inferredInst` parameter and set it locally in the method rather than as a class field.  As a class field, we run the risk of the `speciesAbbreviation` being null or set to an incorrect value if the `setSpeciesAbbreviaiton` method hasn't been called to sync the value with the call to `generateOrthologousStableId`.

Reply:
Or make it an argument to the methods that need it.

Reply:
Yeah, that works too but I would just worry that the inferredInstance and the passed speciesAbbreviation could be out of sync.  The speciesAbbreviation can be derived from the inferredInstance, so passing it as its own parameter isn't really necessary.

Reply:
The speciesAbbreviation comes from the species config file, which already contributes things like the species Name, URL, reference DB, etc. Since the process is called on a per-species basis, if the abbreviation was out of sync that would mean that these other variables are out of sync as well

Reply:
Hmmm good point...

Reply:
Still, if `speciesAbbreviation` is only used in one static method, it's probably  better to just make it an argument to that method. It would mean you don't have to call `setSpeciesAbbreviation` anymore.

Reply:
StableIdentifierGeneration is called from 3 separate classes, so to have it as an argument would be adding more code. Unless there is a strong belief against setting it using the method, I'd like to leave it as is

Reply:
ok

Reply:
It would probably be good to have this class be instantiable and make the static methods into instance methods.  If you want to ensure only one object of this class exists, though, you could use the singleton pattern.

https://www.tutorialspoint.com/design_pattern/singleton_pattern.htm

I'm just concerned that static methods leave it open to being in an inconsistent state.

Reply:
We made this class instantiable so now this abbreviation thing isn't done with a setter